var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -28,9 +28,9 @@ class PyNumba(PythonPackage):
     # That's why it was chosen as an upper bound in the following depends_on
     # calls.  If newer versions maintain backwards compatibility, the calls
     # can be updated accordingly.
-    depends_on('py-llvmlite@0.33:0.34', type=('build', 'run'), when='@0.50.1')
+    depends_on('py-llvmlite@0.33.0', type=('build', 'run'), when='@0.50.1')
     depends_on('py-llvmlite@0.31.0:0.32.0', type=('build', 'run'), when='@0.48.0')
-    depends_on('py-llvmlite@0.25:', type=('build', 'run'), when='@0.40.1:')
+    depends_on('py-llvmlite@0.25:', type=('build', 'run'), when='@0.40.1:0.47.0')
```

Comment:
If the version of numba is not bounded by above on line 32, then line 32 overrides all the depends_on placed above.

Reply:
Good catch.